ldap/servers/plugins/replication/repl5_agmt.c

```diff
@@ -2527,6 +2551,108 @@ agmt_set_consumer_ruv(Repl_Agmt *ra, RUV *ruv)
     return 0;
 }
 
+void
+agmt_update_init_status(Repl_Agmt *ra)
+{
+    int rc;
+    Slapi_PBlock *pb;
+    LDAPMod **mods;
+    int nb_mods = 0;
+    int mod_idx;
+    Slapi_Mod smod_start_time = {0};
+    Slapi_Mod smod_end_time = {0};
+    Slapi_Mod smod_status = {0};
+
+    PR_ASSERT(ra);
+    PR_Lock(ra->lock);
+
+    if (ra->last_init_start_time) {
+        nb_mods++;
+    }
+    if (ra->last_init_end_time) {
+        nb_mods++;
+    }
+    if (ra->last_init_status[0] != '\0') {
+        nb_mods++;
+    }
```

Comment:
```suggestion
    }
    if (nb_mods == 0) {
        PR_Unlock(ra->lock);
        return;
    }
```


I think it's better to leave here if there are no mods. 
It feels a bit strange to check it at the end of the function, and then call `slapi_mod_done(*);` even though no mods were done...

Reply:
Good point. Thanks